src/procstat.c

```diff
@@ -260,12 +260,30 @@ static void fuse_lookup(fuse_req_t req, fuse_ino_t parent_inode, const char *nam
 	}
 
 	fuse_entry.ino = (uintptr_t)item;
+	item->refcnt++;
 	fuse_entry.attr_timeout = ATTRIBUTES_TIMEOUT_SEC;
 	fill_item_stats(context, item, &fuse_entry.attr);
 	pthread_mutex_unlock(&context->global_lock);
 	fuse_reply_entry(req, &fuse_entry);
 }
 
+static void item_put_locked(struct procstat_item *item);
+static void fuse_forget(fuse_req_t req, fuse_ino_t ino, uint64_t nlookup) {
+	struct procstat_context *context = request_context(req);
+	struct procstat_item *item;
+
+	pthread_mutex_lock(&context->global_lock);
+	item = (struct procstat_item *)(ino);
+	if (nlookup >= item->refcnt) {
+		item->refcnt = 1;
+		item_put_locked(item);
```

Comment:
@anton-lb the assumption is that item_put_locked would be done here or in close ?

Reply:
correct